src/core/EdFi.DataManagementService.Core/Middleware/InjectPropertiesToEdFiDocumentMiddleware.cs

```diff
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+using EdFi.DataManagementService.Core.Pipeline;
+using Microsoft.Extensions.Logging;
+
+namespace EdFi.DataManagementService.Core.Middleware
+{
+    internal class InjectPropertiesToEdFiDocumentMiddleware(ILogger _logger) : IPipelineStep
```

Comment:
How about renaming this to `InjectLastModifiedDateMiddleware`? More specific about what it does, and with lots of middlewares specific naming is helpful.